### PR TITLE
feature/close-at-updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** Issue #40 

**This PR will result in the following new package version:** `v0.7.0`

This will be a breaking change since the fields are changing their meaning. If a user was referencing this field previously, the value will likely have changed. As such, this should be treated as a breaking change.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

As discussed within the above linked issue it became apparent that the `first_close_*` and `last_close_*` metrics in the `intercom__conversation_enhanced` and `intercom__conversation_metrics` models was not entirely accurate if an Intercom user is leveraging bots to help auto  close conversations. As such, the following changed/new fields have been applied in this release.

| model | field_name | previous_description | new_description |
|-------|------------|-----------------|-----------------|
|intercom__conversation_enhanced / intercom__conversation_metrics | first_close_at | The time of the first conversation part where part_type was 'close' indicating the conversation was closed. | The time of the first conversation part where part_type was 'close' by any author indicating the conversation was closed. |
|intercom__conversation_enhanced / intercom__conversation_metrics | last_close_at | The time of the last conversation part where part_type was 'close' indicating the conversation was closed. | The time of the last conversation part where part_type was 'close' by any author indicating the conversation was closed. |
|intercom__conversation_enhanced / intercom__conversation_metrics | first_admin_close_at | N/A this is a new field, but was the previous first_close_at value | The time of the first conversation part where part_type was 'close' by an admin indicating the conversation was closed. |
|intercom__conversation_enhanced / intercom__conversation_metrics | last_admin_close_at | N/A this is a new field, but was the previous last_close_at value | The time of the last conversation part where part_type was 'close' by an admin indicating the conversation was closed. |
|intercom__conversation_enhanced / intercom__conversation_metrics | first_close_by_author_id | N/A this is a new field | The author_id of the author (admin, bot, user, etc.) who was first closed to the conversation. |
|intercom__conversation_enhanced / intercom__conversation_metrics | last_close_by_author_id | N/A this is a new field | The author_id of the author (admin, bot, user, etc.) who was last closed to the conversation. |
| intercom__conversation_metrics | time_to_last_close_minutes | Previously leveraged last_close_at which only took into account admin close parts. | The time difference (not factoring in business hours) between the last_close_at and the first_contact_reply. |
| intercom__conversation_metrics | time_to_first_close_minutes | Previously leveraged first_close_at which only took into account admin close parts. | The time difference (not factoring in business hours) between the first_admin_response_at and the first_close_at. |
| intercom__conversation_metrics | time_to_admin_first_close_minutes | N/A this is a new field. | The time difference (not factoring in business hours) between the first_admin_response_at and the first_admin_close_at.|
| intercom__conversation_metrics | time_to_admin_last_close_minutes | N/A this is a new field. | The time difference (not factoring in business hours) between the first_admin_response_at and the last_admin_close_at.| 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt compile
- [X] dbt run –full-refresh
- [X] dbt run
- [X] dbt test
- [X] dbt run –vars (if applicable) 

Tested on a variety of variables to ensure there were no errors across configs

Before marking this PR as “ready for review” the following have been applied:
- [X] The appropriate issue has been linked and tagged
- [X] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as “ready for review”:
- [X] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [X] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [X] You have provided details below around the validation steps performed to gain confidence in these changes.

To validate these changes I inspected the compiled results in the materialized conversation enhanced and metrics models. Within these I was able to confirm the close at values did in fact differ as expected. Additionally, I am sharing with the issue OP to confirm the changes resulted in their expected values. [Results to be shared once validated].

For our data, we did not utilize any bots. Therefore, I was able to validate that the previous and new records all matched (as expected) since there was no one other than the admin who closed tickets. See a quick query that shows this
```sql
select 
  conversation_id,
  first_close_at,
  first_admin_close_at,
  last_close_at,
  last_admin_close_at,
  first_close_by_admin_id,
  last_close_by_admin_id,
  first_close_by_author_id,
  last_close_by_author_id,
  time_to_admin_last_close_minutes,
  time_to_last_close_minutes,
  time_to_admin_first_close_minutes,
  time_to_first_close_minutes
from zz_dbt_joe_intercom_new.intercom__conversation_metrics 
```

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [X] indexed within dbt_project.yml
    - [X] indexed within integration_tests/dbt_project.yml
- [X] CHANGELOG has individual entries for each respective change in this PR
    
- [X] README updates have been applied (if applicable)
    
- [N/A] DECISIONLOG updates have been updated (if applicable)
- [X] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [X] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🌂

### CODE REVIEW

**Comment by fivetran-catfritz** [comment]
Thanks for the explanation it was helpful! The changelog comparisons also clearly laid out what changed. I compared the new outputs against the old, and like you mentioned, I noticed only admins closed tickets, but confirm I see the new columns matched the old columns. I had a documentation suggestion, but otherwise looks good!

**Review by fivetran-catfritz — COMMENTED** [review]
Thanks for the explanation it was helpful! The changelog comparisons also clearly laid out what changed. I compared the new outputs against the old, and like you mentioned, I noticed only admins closed tickets, but confirm I see the new columns matched the old columns. I had a documentation suggestion, but otherwise looks good!

File: models/intercom.yml [hunk 1]
```diff
@@ -188,6 +188,10 @@ models:
         description: The admin_id of the admin who was first assigned to the conversation.
       - name: last_close_by_admin_id
         description: The admin_id of the admin who last closed the conversation.
+      - name: first_close_by_author_id
+        description: The author_id of the author (admin, bot, user, etc.) who was first closed to the conversation.
+      - name: last_close_by_author_id
+        description: The author_id of the author (admin, bot, user, etc.) who was last closed to the conversation.
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intercom.yml]: Good call out. Updated!

File: models/intercom.yml [hunk 2]
```diff
@@ -208,10 +212,14 @@ models:
...
+      - name: first_admin_close_at
+        description: The time of the first conversation part where part_type was 'close' by an admin indicating the conversation was closed.
+      - name: last_admin_close_at
+        description: The time of the last conversation part where part_type was 'close' by an admin indicating the conversation was closed.
       - name: first_close_at
-        description: The time of the first conversation part where part_type was 'close' indicating the conversation was closed.
+        description: The time of the first conversation part where part_type was 'close' by any author indicating the conversation was closed.
       - name: last_close_at
-        description: The time of the last conversation part where part_type was 'close' indicating the conversation was closed.
+        description: The time of the last conversation part where part_type was 'close' by any author indicating the conversation was closed.
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intercom.yml]: Good call out. Updated!

**Comment by fivetran-joemarkiewicz** [comment]
Thanks @fivetran-catfritz! I agree with your suggestions and applied the updates. When you have a moment would you be able to review and provide a final approval if all looks good?

**Review by fivetran-catfritz — APPROVED** [review]
lgtm!

**Pull request merged**